src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
         return indexModel;
     }
 
-    // Minimal
+    /**
+     * Overload for creating a new pod index with environment and podType.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2");
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment, String podType) {
-        validatePodIndexParams(indexName, dimension, environment, podType, null, null, null, null);
-
         return createPodsIndex(indexName, dimension, environment, podType, null, null, null, null, null, null);
     }
 
-    // Minimal + metric
+    /**
+     * Overload for creating a new pod index with environment, podType, and metric.
```

Comment:
Not sure if this matters, but it might help differentiate `metric` from `dimension` if you say `similarity metric` in the docstring.